### ISSUE TITLE
[EPIC] EP06: Dynamic Pricing & Buy Box Optimization

### ISSUE THREAD

**Issue by adaptative** [adaptative/amazon-seller-autopilot#26]
## Epic: Dynamic Pricing & Buy Box Optimization

**Objective:** AI agent that implements intelligent, real-time pricing strategies to win the Buy Box while maintaining target margins.

**Phase:** 1 (Months 3-4)

**Key Outcomes:**
- Real-time competitor price monitoring
- Game-theory pricing to win Buy Box without price wars
- Margin-aware pricing factoring in all fees and costs
- Cross-agent coordination with advertising and inventory

**APIs:** Product Pricing API, Notifications API (ANY_OFFER_CHANGED), Feeds API, Product Fees API

**Comment by Whatsonyourmind** [comment]
Buy Box optimization on Amazon is a sequential decision problem with constraints. The math:

**The explore/exploit tradeoff:** Each price point is an arm. The reward is: `margin * P(win_buy_box | price)`. You need to discover P(win) for different prices while maximizing revenue.

**Thompson Sampling for pricing:**
```python
# For each candidate price point p:
#   Maintain Beta(wins, losses) where:
#   win = got the Buy Box at price p
#   loss = didn't get it at price p

# Each repricing cycle:
samples = {p: Beta(wins[p]+1, losses[p]+1).sample() for p in price_points}
expected_margin = {p: (p - cost) * samples[p] for p in price_points}
selected_price = max(expected_margin, key=expected_margin.get)
```

The beauty: this naturally balances exploration (trying prices you are uncertain about) with exploitation (using the price you know wins) — and it does so in a way that is margin-aware, not just win-rate-aware.

**Constraint layer (margin floor):** Before the bandit selects, filter out price points where `price - cost - fees < min_margin`. This ensures the bandit only explores profitable prices.

**Competitor response modeling (Monte Carlo):** For each candidate price, simulate 1000 scenarios of competitor responses (based on their observed repricing patterns). Compute: P(win_buy_box) = fraction of scenarios where your price wins. This gives you a forward-looking win probability rather than a backward-looking one.

**Cross-ASIN coordination:** If you sell 50 ASINs, run 50 independent bandits. But share the competitor response model across ASINs in the same category — competitor behavior is correlated.
